R/derived-bmi.R

```diff
@@ -0,0 +1,72 @@
+#' Custom ifelse function that evaluates missing (NA) values
+#' 
+#' If the logical argument compares to a value that is NA, it will set `x` to `FALSE`  
+#' 
+#' @param x A logical argument
+#' 
+#' @param a value if `x` is `TRUE`
+#' 
+#' @param b value if `x` is `FALSE`
+#' 
+#' @return None
+#' 
+#' @examples ifelse2((age<18),"child", "adult")
+#' 
+#' @export
+ifelse2 <- function(x, a, b) {
+  falseifNA <- function(x) {
+    ifelse(is.na(x), FALSE, x)
+  }
+  ifelse(falseifNA(x), a, b)
+}
+
+#' Body Mass Index (BMI) derived variable
+#' 
+#' This function creates a derived BMI variable independent from the BMI variable provided by the CCHS. Within the CCHS, 
+#' the method in calculating BMI varies across cycles, leading to measurement error. Therefore, if you would like to 
+#' examine BMI across CCHS cycles it is recommended to use this derived BMI variable. If looking at BMI data in one
+#' CCHS year, it is appropriate to use the CCHS BMI variable (HWTGBMI) that is also available on cchsflow.
+#' 
+#' This variable uses the CCHS variables for height and weight that have been transformed by cchsflow. In order to 
+#' generate a value for BMI across CCHS cycles, height and weight must be transformed and harmonized.
+#' 
+#' In earlier CCHS cycles (2001 and 2003), height was collected in inches; while in later CCHS cycles (2005+) 
+#' it was collected in inches. To harmonize values across cycles, height was converted to meters (to 3 decimal points). 
+#' Weight was collected in kilograms across all CCHS cycles, so no transformations were required in the harmonization process.
+#' 
+#' @param HWTGHTM CCHS variable for height (in meters)
```

Comment:
I prefer this use of the cchflow variable for the function argument. As opposed to the use of new argument names in 
smoking pack years.

What to others think? @DarkShadeKnigh @yulric

Reply:
I wonder about a shorter variable name for derived variables?

`BMI_der` as opposed to `BMI_dereived`

https://github.com/Big-Life-Lab/cchsflow/blob/a1c5448ba094f5a44b19affa5a5f398f0a162b47/R/derived-bmi.R#L44

Reply:
bmi_derived calculates for all ages, whereas some cycles calculated for only age 20 - 64, right? If so, could you add a comment indicating which cycles used which ages?

Reply:
Can you comment on how BMI is calculated for pregnancy? Is it consistent across cycles? Is there consistent NA across cycles?

Reply:
> I prefer this use of the cchflow variable for the function argument. As opposed to the use of new argument names in
> smoking pack years.
> 
> What to others think? @DarkShadeKnigh @yulric

I had originally created new argument names as way to make easier for users to derive complex variables with multiple CCHS variables. But since users will be mainly using the existing derived variables in cchsflow, I agree on sticking to the transformed CCHS variable names.

Reply:
In the params we list which variable corresponds to what and having the non cchs names makes the code much easier to follow. Plus this shows the user that they can use any name they want for ease of their own documentation.

Reply:
@wyusuf068 Explained his point to me further and yeah i agree cchs variable names is better